src/epcpm/sunspectointerface.py

```diff
@@ -0,0 +1,2227 @@
+                        f"{item_name}.common.sunspec{self.sunspec_id.value}.getter(",
+                        [
+                            f"(InterfaceItem_void *) &{item_name},",
+                            f"Meta_Value",
+                        ],
+                        f");",
+                    ]
+                )
+                setter.extend(
+                    [
+                        f"{item_name}.common.sunspec{self.sunspec_id.value}.setter(",
+                        [
+                            f"(InterfaceItem_void *) &{item_name},",
+                            f"true,",
+                            f"Meta_Value",
+                        ],
+                        f");",
+                    ]
+                )
+            else:
+                if getattr(parameter, "sunspec_getter", None) is not None:
+                    getter.append(
+                        parameter.sunspec_getter.format(
+                            interface=sunspec_variable,
+                        )
+                    )
+
+                if getattr(parameter, "sunspec_setter", None) is not None:
+                    setter.append(
+                        parameter.sunspec_setter.format(
+                            interface=sunspec_variable,
+                        )
+                    )
+
+            get_out = epcpm.c.format_nested_lists(getter)
+
+            if not uses_interface_item and not self.wrapped.not_implemented:
+                setter.append(f"{hand_coded_setter_function_name}();")
+
+            if not parameter.read_only:
+                set_out = epcpm.c.format_nested_lists(setter)
+            else:
+                set_out = None
+
+            base_decl = f"SUNSPEC{self.sunspec_id.value}_MODEL{self.model_id}_{table_option}{parameter.abbreviation}"
+
+        return [get_out, set_out, base_decl]
+
+
+@specific_builders(epcpm.sunspecmodel.DataPointBitfield)
+@attr.s
+class SpecificDataPointBitfield:
+    """Specific model interface generator for the SunSpec DataPointBitfield class."""
+
+    wrapped = attr.ib()
+    scale_factor_from_uuid = attr.ib()
+    model_id = attr.ib()
+    is_table = attr.ib()
+    parameter_uuid_finder = attr.ib(default=None)
+    sunspec_id = attr.ib(default=None)
+
+    def gen(self) -> typing.Tuple[typing.List[str], typing.List[str], str]:
+        """
+        Specific model interface generator for the SunSpec DataPointBitfield class.
+
+        Returns:
+            [list[str], list[str], str]: list of getters, list of setters, base declaration string
+        """
+        getter = []
+        setter = []
+
+        if self.wrapped.parameter_uuid is not None:
+            parameter = self.parameter_uuid_finder(self.wrapped.parameter_uuid)
+
+        uses_interface_item = (
+            isinstance(parameter, epyqlib.pm.parametermodel.Parameter)
+            and parameter.uses_interface_item()
+        )
+
+        # TODO: should we just require that it does and assume etc?
```

Comment:
The todos here still relevant? Do they need a jira ticket?

Reply:
Not relevant. b265d71d5b1682a497fa7f7520dbbd5599753550